plugins/rapid7_surface_command/build/lib/icon_rapid7_surface_command/util/surface_command/api_connection.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+import furl
+import requests
+
+
+REGION_MAP = {
+    "United States": "us",
+    "United States 2": "us2",
+    "United States 3": "us3",
+    "Europe": "eu",
+    "Canada": "ca",
+    "Australia": "au",
+    "Japan": "ap",
+}
+
+
+class ApiConnection:
+    """
+    ApiConnection(api_key, region_string, logger)
+
+    A class to connect to the Surface Command API. This class provides convenience methods to perform actions
+    on Surface Command.
+    """
+
+    def __init__(self, api_key: str, region_string: str, logger: logging.Logger) -> None:
+        """
+        Init the connection and set the region
+        """
+        self.api_key = api_key
+        self.logger = logger
+        region = REGION_MAP.get(region_string)
+        self.url = f"https://{region}.api.insight.rapid7.com/surface/graph-api/objects/table"
+
+    def run_query(self, query_id: str) -> dict:
+        """
+        Execute Surface Command Query
+        """
+        url = furl.furl(self.url).set(args={"format": "json"})
+        response = requests.post(
```

Comment:
We might need to surround these and try catch and raise a PluginException from insightconnect_plugin_runtime.exceptions so we can gracefully exit if we get a requests or json error.

We do have a request and json helper inside insightconnect_plugin_runtime.helpers that handles catching these . A recent example of usage [here](https://github.com/rapid7/komand-plugins/blob/f2101d491eeac37f825fbd8c1fd064b185486675/plugins/claroty/icon_claroty/util/api.py#L76).